packages/zk-eddsa-frog-pcd/src/ZKEdDSAFrogPCD.ts

```diff
@@ -0,0 +1,402 @@
+  frogDataToBigInts
+} from "@pcd/eddsa-frog-pcd";
+import type { EdDSAPublicKey } from "@pcd/eddsa-pcd";
+import {
+  ArgumentTypeName,
+  BigIntArgument,
+  DisplayOptions,
+  PCD,
+  PCDArgument,
+  PCDPackage,
+  ProveDisplayOptions,
+  SerializedPCD
+} from "@pcd/pcd-types";
+import {
+  SemaphoreIdentityPCD,
+  SemaphoreIdentityPCDPackage
+} from "@pcd/semaphore-identity-pcd";
+import { STATIC_SIGNATURE_PCD_NULLIFIER } from "@pcd/semaphore-signature-pcd";
+import {
+  fromHexString,
+  generateSnarkMessageHash,
+  hexToBigInt,
+  requireDefinedParameter
+} from "@pcd/util";
+import {
+  Groth16Proof,
+  prove as groth16Prove,
+  verify as groth16Verify
+} from "@zk-kit/groth16";
+import { Eddsa, buildEddsa } from "circomlibjs";
+import JSONBig from "json-bigint";
+import { v4 as uuid } from "uuid";
+import vkey from "../artifacts/circuit.json";
+import { ZKEdDSAFrogCardBody } from "./CardBody";
+
+export const STATIC_ZK_EDDSA_FROG_PCD_NULLIFIER = generateSnarkMessageHash(
+  "hard-coded-zk-eddsa-frog-pcd-nullifier"
+);
+
+/**
+ * The global unique type name of the {@link ZKEdDSAEventTicketPCD}.
+ */
+export const ZKEdDSAFrogPCDTypeName = "zk-eddsa-frog-pcd";
+
+/**
+ * Interface containing the arguments that 3rd parties use to
+ * initialize this PCD package.
+ * These are the artifacts associated with the circom circuit.
+ */
+export interface ZKEdDSAFrogPCDInitArgs {
+  zkeyFilePath: string;
+  wasmFilePath: string;
+}
+
+let initArgs: ZKEdDSAFrogPCDInitArgs | undefined = undefined;
+
+/**
+ * Defines the essential paratmeters required for creating an {@link ZKEdDSAEventTicketPCD}.
+ */
+export type ZKEdDSAFrogPCDArgs = {
+  frog: PCDArgument<EdDSAFrogPCD>;
+
+  identity: PCDArgument<SemaphoreIdentityPCD>;
+
+  externalNullifier: BigIntArgument;
+};
+
+/**
+ * Defines the ZKEdDSAEventTicketPCD claim.
+ */
+export interface ZKEdDSAFrogPCDClaim {
+  frogOmitOwner: Omit<IFrogData, "ownerSemaphoreId">;
+  signerPublicKey: EdDSAPublicKey;
+  externalNullifier: string;
+  nullifierHash: string;
+}
+
+/**
+ * The ZK EdDSA Frog PCD enables the verification that an owner with a Semaphore
+ * identity owns the EdDSA Frog PCD while keeping the owner's semaphore identity private.
+ */
+export class ZKEdDSAFrogPCD
+  implements PCD<ZKEdDSAFrogPCDClaim, Groth16Proof>
+{
+  type = ZKEdDSAFrogPCDTypeName;
+
+  public constructor(
+    readonly id: string,
+    readonly claim: ZKEdDSAFrogPCDClaim,
+    readonly proof: Groth16Proof
+  ) {
+    this.id = id;
+    this.claim = claim;
+    this.proof = proof;
+  }
+}
+
+export async function init(args: ZKEdDSAFrogPCDInitArgs): Promise<void> {
+  initArgs = args;
+}
+
+export function getProveDisplayOptions(): ProveDisplayOptions<ZKEdDSAFrogPCDArgs> {
+  return {
+    defaultArgs: {
+      frog: {
+        argumentType: ArgumentTypeName.PCD,
+        description: "Generate a proof for the selected frog",
+        validate(value, _) {
+          if (value.type !== EdDSAPCDTypeName || !value.claim) {
```

Comment:
```suggestion
          if (value.type !== EdDSAFrogPCDTypeName || !value.claim) {
```

this should be renamed in `packages/eddsa-frog-pcd/src/EdDSAFrogPCD.ts` as well.

Reply:
fixed